Add folder structure to Integration testing page. (#4197)
magento/devdocs: guides/v2.1/test/integration/integration_test_execution.md

Once the system is prepared, the tests can be executed using either the command line interface or within an IDE like PhpStorm.

## Set up the integration test framework

In order to run the integration tests, create and configure a test database.
You might also want to adjust the PHPUnit configuration, depending on your requirements.

Refer to [Preparing Integration Test Execution][setup] for further information about setting up the test environment.

## Command Line Interface (CLI)

This option can be used for running the tests locally during development or on remote servers during Continuous Integration.

Please refer to [Running Integration Tests in the CLI][cli run] for further information.

## PhpStorm IDE

Running the integration tests inside an IDE like PhpStorm IDE is convenient during development. This is mostly used when writing a new integration test.

Other than convenience, there is no benefit over running the tests on the console.

Refer to [Running Integration Tests in PhpStorm][phpstorm run] for further information.

## Integration tests file structure

> The root folder for the Magento integration tests suite is `<magento_root>/dev/tests/integration`.

Reviewer: Perhaps avoid two sentences describing the same folder.

```suggestion
The root folder for the Magento integration tests suite —`<magento_root>/dev/tests/integration`—contains the following sub-folders and files:
```

This folder contains the following sub-folders and files:

- `framework/` – Integration testing framework scripts, configuration files and classes.

> - `Magento/` – A set of classes that actually implement the Magento integration tests framework.

Reviewer: ```suggestion
- `Magento/` – A set of classes that implement the Magento integration tests framework.
```

- `bootstrap.php` – The PHPUnit bootstrap script.

> - `etc/install-config-<db_vendor>.php` – A configuration file that provides values for installing Magento application.

Reviewer: ```suggestion
- `etc/install-config-<db_vendor>.php` – A configuration file that provides values for installing the Magento application.
```

- `testsuite/` – The test suite.

> - `tmp/` – A writable directory for storing temporary data during tests execution.

Reviewer: This line just sounds funny to me right now. "during tests execution"
Not sure whether it is the word execution or if it is the plural tests. Reread just to be sure.

Author: Right. Made singular.

> - `sandbox-<hash>/` – The folder where each Magento instance stores temporary and configuration data.

Reviewer: Temporary data and configuration data? Or temporary configuration data? Maybe I'm reading to too fast.

../../../vendor/bin/phpunit --filter 'testOnlyThisOneIsExecuted' ../../../app/code/Acme/Example/Test/Integration/ExampleTest.php
```

## Common mistakes

> ### Can't read files specified as arguments

Author: Argh.. contraction.

> 

Reviewer: I can't leave a comment on line 324, but it (`Use the integration test configuration file`) could probably be an H3 instead of an H4.

Author: Thanks. Headings were messy on this page.
